services/QuillLMS/spec/queries/staff/rules_analysis_query_spec.rb

```diff
@@ -0,0 +1,175 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+module Staff
+  describe RulesAnalysisQuery do
+    include_context 'QuillBigQuery TestRunner Setup'
+
+    describe '#format' do
+      let(:rule) { create(:evidence_rule, name: 'so_rule1', rule_type: 'autoML') }
+
+      it 'should format correctly' do
+        create(:evidence_feedback, rule: rule, order: 1, text: 'a'*10)
+        create(:evidence_feedback, rule: rule, order: 2, text: 'b'*10)
+
+        example_bigquery_result = [
+          {
+            :id => rule.id,
+            :rules_uid => "0effa491-f9c7-482c-bce4-3982c5644259",
+            :activity_id => 80,
+            :rule_type => "autoML",
+            :rule_suborder => 1,
+            :rule_name => "so_rule2",
+            :rule_note => "This rule is a test",
+            :total_responses => 1,
+            :avg_confidence => nil,
+            :total_strong => 1,
+            :total_weak => 0,
+            :repeated_consecutive => 1,
+            :repeated_non_consecutive => 0
+          }
+      ]
+
+        expected = [{
+          rule_uid: "0effa491-f9c7-482c-bce4-3982c5644259",
+          api_name: rule.rule_type,
+          rule_order: 1,
+          first_feedback: 'a'*10,
+          second_feedback: 'b'*10,
+          rule_note: "This rule is a test",
+          rule_name: "so_rule2",
+          avg_confidence: nil,
+          total_responses: 1,
+          strong_responses: 1,
+          weak_responses: 0,
+          repeated_consecutive_responses: 1,
+          repeated_non_consecutive_responses: 0
+        }]
+
+        expect(
+          described_class.new(activity_id: 1, conjunction: 'so').post_query_transform(example_bigquery_result)
+        ).to eq expected
+      end
+    end
+
+    context 'big_query_snapshot', :big_query_snapshot do
+      context 'without post query transforms' do
+        before do
+          described_class.class_eval { def post_query_transform(data) = data }
```

Comment:
To test BigQuery before running post query transforms, we replace `post_query_transform` with the identity function, for this rspec block.